### ISSUE TITLE
[BUG]: Script that breaks calendar in taskbar

### ISSUE THREAD

**Issue by Julynx** [undergroundwires/privacy.sexy#314]
### Description

I just wanted to know which script is the one that breaks the calendar menu in the taskbar, preventing it from opening when clicking the time/date/bell icon in the taskbar in Windows 11.

I must have applied it by mistake, and when fuzzy searching for terms like Notifications or Action Center, I havent been able to find the correct script to revert it.

### OS

Windows 11 Pro 23H2

### Reproduction steps

I remember I applied at least the Strict settings, plus some extra script I dont remember exactly which tho.

### Scripts

Strict + extra ones I dont remember

### Screenshots

Calendar flashes a bit too fast when I click the date menu but doesnt really stay open to show up in a screenshot. I have recorded a video though, in case it helps:

https://github.com/undergroundwires/privacy.sexy/assets/64993676/10584e96-4ab6-44bb-8504-2c0f72c5a949

**Comment by undergroundwires** [comment]
Hi, did you execute it using desktop application? In that case we can find out your script file to debug this further.

**Comment by oculi38** [comment]
Hi, I'm not sure it's a bug, but I'm glad if it is.

Please see here: https://discuss.privacyguides.net/t/disable-windows-push-notification/19119

**Comment by undergroundwires** [comment]
Hi @oculi38, as I understand from the link you consider **Disable Windows Push Notifications** to be the root cause.

I ran the script on (Disable Windows Push Notifications) on Windows 11 23H2 Pro, but I cannot reproduce this.

Calendar on taskbar and calendar app works fine for me. This is most likely caused by some another script.

We need to find out the exact script that's causing this to find a potential way to do the same thing without breaking stuff, or just document it better.

Do you have the script file you executed? If you used the desktop version, it's [stored locally](https://github.com/undergroundwires/privacy.sexy/blob/master/docs/desktop/desktop-vs-web-features.md#logging) on your computed.

**Comment by oculi38** [comment]
Hi there, I had to rerun privacy.sexy script for some other reasons, so I went through all settings one by one again (I made tweaks to the 'Strict' option). I saved the script for future use, including the script "Disable Windows Push Notifications" under the sections 'Remove bloatware' -> 'Disable non-essential services'. The scripts broke the calendar + notification area (bottom right corner of Windows 11).

I confirm that by reverting that option, it worked again.

**Comment by oculi38** [comment]
Hi @undergroundwires, were you able to reproduce at all?

**Comment by undergroundwires** [comment]
Hi, I may have figured this out.

- "Strict"  does not cause this. 
- "Disable non-essential services" category causes this.

I tested every service in that category and found out that `Disable Windows Push Notifications` breaks taskbar.

Can you try reverting it and verify that the calendar is back?

**Comment by undergroundwires** [comment]
You are right:

- "Strict"  does not cause this. 
- `Disable Windows Push Notifications` causes this.

It started breaking after restarting Windows 11 23H2 Pro. It works fine on Windows 10 versions (does not break taskbar).

Does reverting that solve the issue for you?

**Comment by oculi38** [comment]
Yes reverting solved the issue, but was there a way to still disable that telemetry without breaking the taskbar?

**Comment by undergroundwires** [comment]
There has also been issues like #110 and #227. Both unresolved. I want  to focus on this after releasing a patch in following days:

I plan to do this:

1. Research alternative ways to get rid of this functionality, add them.
2. Research which versions this script break exactly, and if the script gets executed on these versions, stop its execution (add Windows version constraints).

I'm not sure how to approach #110 though, I'd happy if you can comment whether we should have this or not under the issue.

**Comment by undergroundwires** [comment]
Thank you for the report and sorry for my late response.
I will remove disabling `WpnUserService` for Windows later than 19H2 in next patch as a quick action.
Later, I will research alternative ways to get rid of this functionality, add them in privacy.sexy.

I'm not sure about #225 though and would appreciate any feedback/opinion.

---

Confirmed side effects per Windows version after disabling `WpnUserService` and rebooting:

| Windows Version | Taskbar Notifications Center (#314) | Notifications & Actions Settings (#227) | Network Settings (#110) |
| --------------- | ---------------------------- | -------------------------------- |------------------|
| Windows 11 23H2 | 🔴 Affected | 🔴 Affected | 🟢 Unaffected |
| Windows 11 22H2 | 🔴 Affected | 🔴 Affected | 🟢 Unaffected |
| Windows 11 21H2 | 🔴 Affected | 🟢 Unaffected | 🟢 Unaffected |
| Windows 10 22H2 | 🟢 Unaffected | 🟢 Unaffected | 🔴 Breaks |
| Windows 10 21H2 | 🟢 Unaffected | 🟢 Unaffected | 🔴 Breaks |
| Windows 10 20H2 | 🟢 Unaffected | 🟢 Unaffected | 🔴 Breaks |
| Windows 10 19H2 | 🟢 Unaffected | 🟢 Unaffected | 🟢 Unaffected |
| Windows 10 19H1 | 🟢 Unaffected | 🟢 Unaffected | 🟢 Unaffected |

**Comment by undergroundwires** [comment]
This is well documented now and the script does no longer disable `WpnUserService` after 19H2 since [0.13.6](https://github.com/undergroundwires/privacy.sexy/releases/tag/0.13.6) 🚀

**Comment by oculi38** [comment]
So the script "disable windows push notifications" is still there, but doesn't break the calendar anymore?

**Comment by undergroundwires** [comment]
No it does not, see commit 50ba00b0af6232fc9187532635b04c4d9d9a68af for more details.
I'm not closing the issue as I may have found some other ways to disable cloud-based notifications more reliably. I'm still researching.
